File: tests/src/test/java/com/optivem/atddaccelerator/templategenerator/systemtest/dsl/github/helpers/PackageClient.java
package com.optivem.atddaccelerator.templategenerator.systemtest.dsl.github.helpers;

import com.optivem.atddaccelerator.templategenerator.systemtest.clients.GithubClient;
import com.optivem.atddaccelerator.templategenerator.systemtest.util.Language;

public class PackageClient {
    private final GithubClient client;
    private final String repositoryPath;

    public PackageClient(GithubClient client) {
        this.client = client;
        this.repositoryPath = client.getRepositoryPath();
    }

    public void verifyPackagesExist(Language systemLanguage) {
        for(var l : Language.getAll()) {
            var path = String.format("%s/monolith-%s", repositoryPath, l.getValue());
            if(l.equals(systemLanguage)) {
                verifyPackageExists(path);
            } else {
                verifyPackageDoesNotExist(path);
            }
        }
    }

    private void verifyPackageDoesNotExist(String path) {
    }

    private void verifyPackageExists(String path) {
        
    }
}

File: console/tests/Dsl/GitHub/Helpers/PackageClient.cs
using Optivem.AtddAccelerator.TemplateGenerator.SystemTests.Clients;
using Optivem.AtddAccelerator.TemplateGenerator.SystemTests.Util;

namespace Optivem.AtddAccelerator.TemplateGenerator.SystemTests.Dsl.GitHub.Helpers
{
    public class PackageClient
    {
        private readonly GithubClient _client;

        public PackageClient(GithubClient client)
        {
            _client = client;
        }

        public void VerifyPackagesExist(Language systemLanguage)
        {
            // TODO: Implement package verification logic
            // This would verify that GitHub Packages are created correctly
        }
    }
}